.github/workflows/changelog-auto-add.yml

```diff
@@ -1,10 +1,48 @@
 name: 'Changelog Auto Add'
-on: workflow_dispatch
+on:
+    pull_request:
+        types: [opened, synchronize, reopened, edited]
+    workflow_dispatch:
+        inputs:
+            prNumber:
+                description: Pull request number
+                required: true
 
 jobs:
-    hello-world:
-        name: 'Hello World'
+    add-changelog:
+        name: 'Add changelog to PR'
+        if: ${{ github.event.pull_request.user.login != 'github-actions[bot]' }}
         runs-on: ubuntu-20.04
+        permissions:
+            contents: write
+            issues: write
+            pull-requests: write
         steps:
-            - name: Hello
-              run: echo "Hello World"
+            - name: Checkout code
+              uses: actions/checkout@v3
+              with:
+                  fetch-depth: 0
```

Comment:
This is intentional ay because you want all the commits from the whole PR?

Not necessary change to make but for future it would be cool to explore just fetching commits for the PR, cause this fetches all branches and tags.

I saw something like this with a quick google:

```
- name: 'PR commits + 1'
  run: echo "PR_FETCH_DEPTH=$(( ${{ github.event.pull_request.commits }} + 1 ))" >> "${GITHUB_ENV}"

- name: 'Checkout PR branch and all PR commits'
  uses: actions/checkout@v3
    with:
      ref: ${{ github.event.pull_request.head.ref }}
      fetch-depth: ${{ env.PR_FETCH_DEPTH }}
 ```

Reply:
>This is intentional ay because you want all the commits from the whole PR?

No, good catch. When the CLI tool clones the repo, yes there needs to be a commit history for calculation the diff but not on this initial checkout. Fixed in https://github.com/woocommerce/woocommerce/pull/38510/commits/047ff513e38744b8fc352e6a58f6a11d21b12a81